src/crypto/abe/interfaces/policy.ts

```diff
@@ -0,0 +1,156 @@
+/* tslint:disable:max-classes-per-file */
+import { logger } from 'utils/logger'
+
+export class PolicyAxis {
+  private _name: string
+  private _attributes: string[]
+  private _hierarchical: boolean
+
+  // Getters and setters
+  public get attributes (): string[] {
+    return this._attributes
+  }
+
+  public set attributes (value: string[]) {
+    this._attributes = value
+  }
+
+  public get name (): string {
+    return this._name
+  }
+
+  public set name (value: string) {
+    this._name = value
+  }
+
+  public get hierarchical (): boolean {
+    return this._hierarchical
+  }
+
+  public set hierarchical (value: boolean) {
+    this._hierarchical = value
+  }
+
+  // Constructor
+  constructor (name: string, attributes: string[], hierarchical: boolean) {
+    this._name = name
+    this._attributes = attributes
+    this._hierarchical = hierarchical
+  }
+}
+
+export class Policy {
+  private readonly _axis: PolicyAxis[]
+  private readonly _maxAttributeCreations: number
+  private readonly _lastAttributeValue?: number
+  private readonly _attributeToInt?: {}
+
+  constructor (axis: PolicyAxis[], maxAttributeCreations: number, lastAttributeValue?: number, attributeToInt?: {}) {
+    this._axis = axis
+    this._maxAttributeCreations = maxAttributeCreations
+    this._lastAttributeValue = lastAttributeValue
+    this._attributeToInt = attributeToInt
+  }
+
+  // Convert this `Policy` to JSON. Output example:
+  // {
+  //     "last_attribute_value": 10,
+  //     "max_attribute_creations": 100,
+  //     "axes": {
+  //         "Security Level": [
+  //         [
+  //             "Protected",
+  //             "Low Secret",
+  //             "Medium Secret",
+  //             "High Secret",
+  //             "Top Secret"
+  //         ],
+  //         true
+  //         ],
+  //         "Department": [
+  //         [
+  //             "R&D",
+  //             "HR",
+  //             "MKG",
+  //             "FIN"
+  //         ],
+  //         false
+  //         ]
+  //     },
+  //     "attribute_to_int": {
+  //         "Security Level::Low Secret": [
+  //         2
+  //         ],
+  //         "Department::MKG": [
+  //         8
+  //         ],
+  //         "Security Level::Medium Secret": [
+  //         3
+  //         ],
+  //         "Security Level::Top Secret": [
+  //         5
+  //         ],
+  //         "Security Level::Protected": [
+  //         1
+  //         ],
+  //         "Department::FIN": [
+  //         10,
+  //         9
+  //         ],
+  //         "Department::HR": [
+  //         7
+  //         ],
+  //         "Department::R&D": [
+  //         6
+  //         ],
+  //         "Security Level::High Secret": [
+  //         4
+  //         ]
+  //     }
+  // }
+  public toJsonEncoded (): Uint8Array {
+    const policy: any = {}
+    policy.axes = {}
+    policy.attribute_to_int = {}
+    if (this._lastAttributeValue === undefined && this._lastAttributeValue === undefined) {
```

Comment:
I think we should rather do this in Rust: we should have a wasm interface in the `abe_policy` crate to convert a policy into Json. It is too easy to introduce incompatibilities if two crates have their own implementation of the serialization.